torch/utils/data/dataloader.py

```diff
@@ -774,6 +774,27 @@ def _try_get_data(self, timeout=_utils.MP_STATUS_CHECK_INTERVAL):
                 raise RuntimeError('DataLoader worker (pid(s) {}) exited unexpectedly'.format(pids_str))
             if isinstance(e, queue.Empty):
                 return (False, None)
+            import tempfile
+            import errno
+            try:
+                # If we are here, it might be because the FDs we were about to
+                # receive over a socket would exceed the open FDs limit. Let's
+                # test it the obvious way.
+
+                # Raise an exception if we are this close to the FDs limit.
+                # Apparently, trying to open only one file is not a sufficient
+                # test but I don't understand why.
+                fds_limit_margin = 10
```

Comment:
The reason you need more than 1 could be that you need multiple FDs to receive that message from the socket. I imagine that it'd be impossible to get the correct # of FD needed. Maybe 10 is okay.